core/corehttp/gateway_handler_unixfs.go

```diff
@@ -30,9 +140,124 @@ func (i *gatewayHandler) serveUnixFs(w http.ResponseWriter, r *http.Request, res
 	// Handling Unixfs directory
 	dir, ok := dr.(files.Directory)
 	if !ok {
-		internalWebError(w, fmt.Errorf("unsupported UnixFs type"))
+		internalWebError(w, fmt.Errorf("unsupported Unixfs type"))
 		return
 	}
 	logger.Debugw("serving unixfs directory", "path", contentPath)
 	i.serveDirectory(w, r, resolvedPath, contentPath, dir, begin, logger)
 }
+
+// redirect returns redirected, newPath (if rewrite), error
+func (i *gatewayHandler) redirect(w http.ResponseWriter, r *http.Request, path ipath.Resolved) (bool, string, error) {
```

Comment:
I haven't touched any of this redirect file parsing logic yet.  Will need to settle on a format, ensure we handle optional bits like status code, etc.  The example file I've been using is...

```
# redirects file
test    /         301
^/hir$  /hi.html  302
^/hi    /hi.html  200
```

Reply:
Good idea.  I'm guessing at the very least we'll have to contribute to another library to get feature parity with what Netlify currently supports (and I'm assuming that's our target to start with), but I'd rather not completely reinvent the wheel.